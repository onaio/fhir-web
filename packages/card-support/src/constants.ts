// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
