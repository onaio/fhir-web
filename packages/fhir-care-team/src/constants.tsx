// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
