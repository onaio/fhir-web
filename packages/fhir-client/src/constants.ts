export const dateFormat = 'yyyy-MM-dd';
