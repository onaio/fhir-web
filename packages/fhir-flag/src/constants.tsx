// urls
export const URL_ADMIN = '/admin';
export const URL_CLOSE_FLAGS = `${URL_ADMIN}/close-flags`;

// // magic strings

export const thatiMinutes = 30 * 60 * 1000;

// form fields
export const locationName = 'locationName' as const;
export const productName = 'productName' as const;
export const status = 'status' as const;
export const comments = 'comments' as const;
