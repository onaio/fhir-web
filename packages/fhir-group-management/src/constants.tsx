// url
export const ADD_EDIT_GROUP_URL = '/group/add-edit';
export const ADD_EDIT_COMMODITY_URL = '/commodity/add-edit';
export const LIST_GROUP_URL = '/groups/list';
export const LIST_COMMODITY_URL = '/commodity/list';

// magic strings
export const groupResourceType = 'Group';
export const listResourceType = 'List';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const type = 'type' as const;
export const unitOfMeasure = 'unitOfMeasure' as const;
