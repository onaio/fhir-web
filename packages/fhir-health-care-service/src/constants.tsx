// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTH_CARE_SERVICE_RESOURCE_TYPE = 'HealthcareService';
export const HEALTHCARES_ENDPOINT = 'HealthcareService/';
