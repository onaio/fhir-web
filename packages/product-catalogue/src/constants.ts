// opensrp api strings
export const OPENSRP_API_BASE_URL = 'https://mg-eusm-staging.smartregister.org/opensrp/rest/';
export const OPENSRP_PRODUCT_CATALOGUE = 'product-catalogue';
export const PRODUCT_ID_ROUTE_PARAM = 'productId';

// router routes
export const CATALOGUE_LIST_VIEW_URL = '/product-catalogue/list';
export const CATALOGUE_EDIT_VIEW_URL = '/product-catalogue/edit';
export const CATALOGUE_CREATE_VIEW_URL = '/product-catalogue/new';
export const HOME_URL = '/';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'product-catalogue';

/** route params for product-catalogue pages */
export interface RouteParams {
  productId?: string;
}
