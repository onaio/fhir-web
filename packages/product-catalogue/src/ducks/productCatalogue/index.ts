/** Redux module for product-catalogue
 * initially developed for eusm-web
 */

import {
  fetchActionCreatorFactory,
  removeActionCreatorFactory,
  setTotalRecordsFactory,
  reducerFactory,
  getItemsByIdFactory,
  getItemsArrayFactory,
  getItemByIdFactory,
  getTotalRecordsFactory,
} from '@opensrp/reducer-factory';

/** interface for a single Product catalogue as received from the api **/
export interface ProductCatalogue {
  productName: string;
  materialNumber: string;
  isAttractiveItem: boolean;
  condition: string;
  appropriateUsage: string;
  accountabilityPeriod: number;
  serverVersion: number;
  uniqueId: number;
  availability: string;
  productPhoto?: unknown;
}

/** reducer name for the Item module */
export const reducerName = 'product-catalogue';

/** Action types */
const FETCHED_CATALOGUE_TYPE = `${reducerName}/LOCATION_TAGS_FETCHED`;
const REMOVE_CATALOGUE_TYPE = `${reducerName}/REMOVE_LOCATION_TAGS`;
const SET_TOTAL_RECORDS_TYPE = `${reducerName}/SET_TOTAL_LOCATION_TAGS`;

/** Item Reducer */
export const ProductCatalogueReducer = reducerFactory<ProductCatalogue>(
  reducerName,
  FETCHED_CATALOGUE_TYPE,
  REMOVE_CATALOGUE_TYPE,
  SET_TOTAL_RECORDS_TYPE
);

// action
export const fetchProducts = fetchActionCreatorFactory<ProductCatalogue>(reducerName, 'uniqueId');
export const removeProducts = removeActionCreatorFactory(reducerName);
export const setTotalProducts = setTotalRecordsFactory(reducerName);

// selectors
export const getProductsById = getItemsByIdFactory<ProductCatalogue>(reducerName);
export const getProductById = getItemByIdFactory<ProductCatalogue>(reducerName);
export const getProductArray = getItemsArrayFactory<ProductCatalogue>(reducerName);
export const getTotalProducts = getTotalRecordsFactory(reducerName);
