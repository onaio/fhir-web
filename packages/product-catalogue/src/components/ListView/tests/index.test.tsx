import React from 'react';
import { ConnectedProductCatalogueList } from '..';
import { store } from '@opensrp/store';
import { createBrowserHistory } from 'history';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { product1, products } from '../../../ducks/productCatalogue/tests/fixtures';
import { CATALOGUE_LIST_VIEW_URL } from '../../../constants';
import { Helmet } from 'react-helmet';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import { removeProducts } from '../../../ducks/productCatalogue';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

describe('List view Page', () => {
  afterEach(() => {
    fetch.resetMocks();
  });

  beforeEach(() => {
    store.dispatch(removeProducts());
  });

  it('renders correctly', async () => {
    fetch.mockResponse(JSON.stringify([]));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: {},
        path: `${CATALOGUE_LIST_VIEW_URL}`,
        url: `${CATALOGUE_LIST_VIEW_URL}`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    /** loading view */
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Loading...Fetching product CataloguePlease wait, as we fetch the product Catalogue."`
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    // no data
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Product Catalogue (0) + Add product to catalogueProduct NameIDActionsNo Data"`
    );

    // details view is not displayed
    expect(wrapper.find('.view-details-content')).toHaveLength(0);
  });

  it('sort works', async () => {
    const product3 = { ...product1, productName: 'Scale', uniqueId: '3' };
    fetch.mockResponse(JSON.stringify([...products, product3]));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: {},
        path: `${CATALOGUE_LIST_VIEW_URL}`,
        url: `${CATALOGUE_LIST_VIEW_URL}`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    // check if page title is correct
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual('Product Catalogue (3)');

    // details view is not displayed
    expect(wrapper.find('.view-details-content')).toHaveLength(0);

    // find ant table
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`table rows ${index}`);
    });

    // click on sort twice to change the order
    wrapper.find('thead tr th').first().simulate('click');
    wrapper.update();

    // click on sort twice to change the order
    wrapper.find('thead tr th').first().simulate('click');
    wrapper.update();

    // check new sort order
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows ${index}`);
    });
  });

  it('renders correctly with detailView', async () => {
    fetch.mockResponse(JSON.stringify(products));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}/1`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: { productId: '1' },
        path: `${CATALOGUE_LIST_VIEW_URL}/:productId`,
        url: `${CATALOGUE_LIST_VIEW_URL}/1`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    /** loading view */
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Loading...Fetching product CataloguePlease wait, as we fetch the product Catalogue."`
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    // find ant table
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`table rows ${index}`);
    });

    // details view is displayed
    expect(wrapper.find('.view-details-content')).toHaveLength(2);

    expect(wrapper.find('div.view-details-content').text()).toMatchSnapshot('View details');
  });

  it('shows broken page', async () => {
    fetch.mockReject(new Error('Something went wrong'));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}/6`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: { productId: '1' },
        path: `${CATALOGUE_LIST_VIEW_URL}/:productId`,
        url: `${CATALOGUE_LIST_VIEW_URL}/6`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    /** loading view */
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Loading...Fetching product CataloguePlease wait, as we fetch the product Catalogue."`
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    /** error view */
    expect(wrapper.text()).toMatchInlineSnapshot(`"ErrorSomething went wrongGo backGo home"`);
  });
});
