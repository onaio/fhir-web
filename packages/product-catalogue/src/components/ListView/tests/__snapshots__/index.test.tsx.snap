// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List view Page renders correctly with detailView: View details 1`] = `"Product nameScaleUnique ID1Material numberMT-123Server version123456"`;

exports[`List view Page renders correctly with detailView: table rows 0 1`] = `"Product NameIDActions"`;

exports[`List view Page renders correctly with detailView: table rows 1 1`] = `"Midwifery Kit2EditView details"`;

exports[`List view Page renders correctly with detailView: table rows 2 1`] = `"Scale1EditView details"`;

exports[`List view Page sort works: sorted table rows 0 1`] = `"Product NameIDActions"`;

exports[`List view Page sort works: sorted table rows 1 1`] = `"Scale1EditView details"`;

exports[`List view Page sort works: sorted table rows 2 1`] = `"Scale3EditView details"`;

exports[`List view Page sort works: sorted table rows 3 1`] = `"Midwifery Kit2EditView details"`;

exports[`List view Page sort works: table rows 0 1`] = `"Product NameIDActions"`;

exports[`List view Page sort works: table rows 1 1`] = `"Midwifery Kit2EditView details"`;

exports[`List view Page sort works: table rows 2 1`] = `"Scale1EditView details"`;

exports[`List view Page sort works: table rows 3 1`] = `"Scale3EditView details"`;
